### PR TITLE
Render comment error and create UserGeneral and UserLeadership components

### LINKED ISSUE #802: Set up error rendering for comment creation (student)

### LINKED ISSUE #815: Create user card children components (user general and user leadership)
General Information:
```
first_name
last_name
email
```

Leadership Information:
```
leadership_type (can be n/a)
group (can be n/a)
```

### PR DESCRIPTION
**Pull request by unzunz**
Fix #802 
Fix #815

### CODE REVIEW

**Comment by warrenshen** [comment]
Fix the pull request title when you get a chance!

File: app/assets/javascripts/actions/student_actions.es6.jsx [hunk 1]
```diff
@@ -15,10 +15,12 @@
       attributes[template.key] = template.value;
       var params = { comment: attributes };
       var resolve = (response) => this.storeComment(response);
+      var reject = (response) => this.storeError(response);
```
Comment (warrenshen) [inline app/assets/javascripts/actions/student_actions.es6.jsx]: Beautiful!

File: app/assets/javascripts/components/user/user_general.es6.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class UserGeneral extends UserCard {
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_general.es6.jsx]: I don't think this should extend `UserCard`?

File: app/assets/javascripts/components/user/user_card.es6.jsx [hunk 3]
```diff
@@ -31,25 +32,15 @@ class UserCard extends Component {
...
-            value={user.email} />
-        </div>
+      <div style={StyleConstants.containers.grid}>
+        <UserGeneral
+          editable={this.props.editable}
+          media={this.props.media}
+          user={this.props.user} />
+        <UserLeadership
+          editable={this.props.editable}
+          media={this.props.media}
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_card.es6.jsx]: Don't need to pass down `media

File: app/assets/javascripts/stores/user_store.es6.jsx [hunk 4]
```diff
@@ -5,6 +5,7 @@
       this.overlay = false;
       this.user = {
         responsibilities: [],
+        leadership: {},
```
Comment (warrenshen) [inline app/assets/javascripts/stores/user_store.es6.jsx]: `l` before `r`!

File: app/assets/javascripts/components/user/user_grid.es6.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  // Render
+  // --------------------------------------------------
+  render() {
+    return (
+      <div style={StyleConstants.containers.grid}>
+        <UserCard
+          editable={this.props.editable}
+          media={this.props.media}
+          user={this.props.user}
+          type={TypeConstants.user.general} />
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_grid.es6.jsx]: `type` before `user`!

File: app/assets/javascripts/components/user/user_leadership.es6.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    };
+  }
+
+  // --------------------------------------------------
+  // Render
+  // --------------------------------------------------
+  renderGroup() {
+    var leadership = this.props.user.leadership;
+    if (leadership) {
+      if (leadership.group) {
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_leadership.es6.jsx]: A `leadership` should always have a `group`, so I think you can refactor this conditional out!

File: app/assets/javascripts/components/user/user_leadership.es6.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      );
+    }
+  }
+
+  renderLeadership() {
+    var leadership = this.props.user.leadership;
+    if (leadership) {
+      return (
+        <CardAttribute
+          label={'Leadership Type'}
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_leadership.es6.jsx]: For consistency, let's do `Leadership type`!

File: app/assets/javascripts/components/user/user_leadership.es6.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          value={leadership.style} />
+      );
+    } else {
+      return (
+        <CardAttribute
+          label={'Leadership Type'}
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_leadership.es6.jsx]: Same here.

File: app/assets/javascripts/constants/type_constants.es6.jsx [hunk 9]
```diff
@@ -48,6 +48,13 @@
       };
     }
 
+    get user() {
+      return {
+        general: 'user-generap',
```
Comment (warrenshen) [inline app/assets/javascripts/constants/type_constants.es6.jsx]: `user-general`.

**Comment by warrenshen** [comment]
A few comments @unzunz!

File: app/assets/javascripts/components/user/user_leadership.es6.jsx (hunk 6 above)
Comment (unzunz) [inline app/assets/javascripts/components/user/user_leadership.es6.jsx]: Getting an error when I take the conditional out:
`Uncaught TypeError: Cannot read property 'full_name' of undefined`

Comment (unzunz) [inline app/assets/javascripts/components/user/user_leadership.es6.jsx]: Getting an error when I take the conditional out
`Uncaught TypeError: Cannot read property 'full_name' of undefined`

**Pull request merged**